test/jdk/java/nio/channels/DatagramChannel/Connect.java

```diff
@@ -48,84 +50,78 @@ static void test() throws Exception {
         invoke(a, r);
     }
 
-    static void invoke(Sprintable reader, Sprintable writer) throws Exception {
-
-        Thread writerThread = new Thread(writer);
-        writerThread.start();
-
-        Thread readerThread = new Thread(reader);
-        readerThread.start();
-
-        writerThread.join();
-        readerThread.join();
-
-        reader.throwException();
-        writer.throwException();
+    static void invoke(Runnable reader, Runnable writer) throws CompletionException {
+        CompletableFuture<Void> f1 = CompletableFuture.runAsync(writer);
```

Comment:
These run the reader and write in the common pool, are you sure that is what you want?

Reply:
> These run the reader and write in the common pool, are you sure that is what you want?

It shouldn't matter since there's no security manager. Are you worried that the common pool may not have enough threads available to run these two tasks concurrently?

Reply:
These are blocking tasks. It's easy to create a thread pool in this test.